[docs] Improved the main README of the SRT project (#2357)
Haivision/srt: docs/build/build-linux.md

# Building SRT for Linux

> Install cmake and openssl-devel (or similar name) package. For pthreads
> there should be -lpthreads linker flag added.

Reviewer: ```suggestion
Install `cmake` and `openssl-devel` (or equivalent) packages. For `pthreads`, add
the `-lpthreads` linker flag.
```

# Building SRT on Linux

Install `cmake` and `openssl-devel` (or equivalent) packages. For `pthreads`, add

> the `-lpthreads` linker flag.

Reviewer 2: I don't think you have to do this thing manually - it should be cleared up by cmake.

Author: @ethouris At this point, I haven't had a task to improve build instructions. I've just moved the text from README to a separate file. All the build instructions require revision and improvements. If you could help me with some of them would be nice. We can start with Linux and macOS. Windows is there, no need to touch. Everything else was written by the community members. Something to revise at the latest mile or ask community to help with.

Currently, we are wrapping up for release. So this is something that could be done right after release. If you have time and desire, please ping me.

Reviewer 2: I understand, I just notified that as it popped up. Worth to be fixed, possibly in another PR.
